### DISCUSSION TITLE
Alignment of cropped tileimage at center of view.

### DISCUSSION THREAD

**Discussion by prasadrc** [openseadragon/openseadragon#2228] [Q&A]
Hi
I want to crop tileimage and I want to display cropped region at the center of the view.So to crop "setCroppingPolygons(polygon)" method  I have used and fitbound method to fit the view but it is not displaying at the center of the view.Can anyone suggest how can I achieve this.

**Comment by iangilman** [comment]
What rectangle are you using for `fitBounds`? The rectangle should be the cropped area, in viewport coordinates. I assume you're using `setCroppingPolygons` because you need a non-rectangular crop? If it's just rectangular, you can use `setClip` instead, and then use `getClippedBounds` to get the correct rectangle. 

I don't think there's an analogous function for cropping polygons, but you can figure it out by running through all of the points in the polygon and finding the min and max x and y values. That'll be the correct rectangle in image coordinates. You can then use the TiledImage's `imageToViewportRectangle` to convert to viewport coordinates.

**Reply by prasadrc** [reply]
Thanks @iangilman my issue got resolved I used setclip() .

**Reply by iangilman** [reply]
Excellent!
